src/modules/ChannelList/dux/reducers.js

```diff
@@ -42,63 +43,70 @@ export default function reducer(state, action) {
         ],
       };
     }
-    case actions.USER_INVITED: {
-      const channel = action.payload;
-      if (state.channelListQuery) {
-        if (filterChannelListParams(state.channelListQuery, channel, state.currentUserId)) {
-          return {
-            ...state,
-            allChannels: getChannelsWithUpsertedChannel(state.allChannels, channel),
-          };
-        }
-        return {
-          ...state,
-          currentChannel: channel,
-        };
-      }
-      return {
-        ...state,
-        allChannels: [channel, ...state.allChannels.filter((ch) => ch.url !== channel?.url)],
-      };
-    }
     case actions.CREATE_CHANNEL: {
       const channel = action.payload;
-      if (state.channelListQuery) {
-        if (filterChannelListParams(state.channelListQuery, channel, state.currentUserId)) {
+      const {
+        allChannels,
+        currentUserId,
+        channelListQuery,
+      } = state;
+      if (channelListQuery) {
+        if (filterChannelListParams(channelListQuery, channel, currentUserId)) {
+          // Good to add to the ChannelList
           return {
             ...state,
-            allChannels: getChannelsWithUpsertedChannel(state.allChannels, channel),
+            currentChannel: channel,
+            allChannels: getChannelsWithUpsertedChannel(allChannels, channel),
           };
         }
+        // Do not add to the ChannelList
         return {
           ...state,
           currentChannel: channel,
```

Comment:
no need to add allChannels here?

Reply:
Yes, because the channel is not appropriate, it's filtered